ciao-storage/ceph.go

```diff
@@ -32,31 +32,67 @@ type CephDriver struct {
 }
 
 // CreateBlockDevice will create a rbd image in the ceph cluster.
-func (d CephDriver) CreateBlockDevice(imagePath *string, size int) (BlockDevice, error) {
-	// generate a UUID to use for this image.
-	ID := uuid.Generate().String()
+func (d CephDriver) CreateBlockDevice(volumeUUID string, imagePath string, size int) (BlockDevice, error) {
+	if volumeUUID == "" {
+		volumeUUID = uuid.Generate().String()
+	}
```

Comment:
We should probably check supplied, i.e., not "",  UUIDs are valid by calling the uuid.Parse function.  I'm just wondering what would happen if someone included a '@' in a UUID.